scenes/globals/pause.gd

```diff
@@ -0,0 +1,114 @@
+extends Node
+
+enum System {
+	PLAYER_INPUT,
+	GAME
+	# For now is just these two but in the future we could have: Physics, Enemies, AI, etc
+}
+
+# Private vars
+var _pause_requests: Array[PauseRequest] = []
+
+# This should be an array of systems but it throws
+# a runtime error when using the filter function, so for now just "Array"
+var _paused_systems: Array = []
+
+
+# Public interface
+func pause_system(system: System, node: Node) -> void:
+	_add_request(system, node)
+
+
+func unpause_system(system: System, node: Node) -> void:
+	_clear_request(system, node)
+
+	_refresh_systems()
+
+
+func is_paused(system: System) -> bool:
+	return system in paused_systems()
+
+
+func paused_systems() -> Array:
+	return _paused_systems
+
+
+func unpaused_systems() -> Array:
+	return System.values().filter(func(system: System) -> bool: return !is_paused(system))
+
+
+# Private methods
+func _find_request(a_system: System, a_node: Node) -> PauseRequest:
```

Comment:
If `PauseRequest` is replaced by a Dictionary, you could just use `Dictionary.get()` or `Dictionary.has()` instead of this private method.

Reply:
Yes! I'll see if I can implement the Dictionary of Arrays